internal/jimm/juju/controller.go

```diff
@@ -695,15 +701,27 @@ func (j *JujuManager) InitiateMigration(ctx context.Context, user *openfga.User,
 		}
 	}
 
-	model := dbmodel.Model{}
-	model.SetTag(mt)
-	err = j.Database.GetModel(ctx, &model)
+	migrationMode := dbmodel.MigrationModeExporting
+	if internalMigration {
+		migrationMode = dbmodel.MigrationModeMigrateInternal
+	}
+	model, err := j.Database.SetModelMigrationMode(ctx, mt.Id(), migrationMode)
 	if err != nil {
-		return result, errors.E(op, "failed to retrieve the model from the database", err)
+		return result, errors.E(op, fmt.Errorf("failed to set model as migrating: %v", err))
+	}
+
+	// Until we have better handling for partial failures we try to revert
+	// the model migration mode if we fail after this to avoid inconsistenties.
+	rollbackMigrationMode := func() {
```

Comment:
Call this "setMigrationModeNone"? I actually think we should just have a db method for this to be honest. You're essentially defining one anyway.